src/lib/network_pool/intf.ml

```diff
@@ -0,0 +1,113 @@
+open Async_kernel
+open Core_kernel
+open Coda_base
+open Pipe_lib
+
+module type Resource_pool_base_intf = sig
+  type t [@@deriving sexp]
+
+  type transition_frontier
+
+  val create :
+       logger:Logger.t
+    -> trust_system:Trust_system.t
+    -> frontier_broadcast_pipe:transition_frontier Option.t
+                               Broadcast_pipe.Reader.t
+    -> t
+end
+
+module type Resource_pool_diff_intf = sig
+  type pool
+
+  type t [@@deriving sexp]
+
+  val summary : t -> string
+
+  val apply : pool -> t Envelope.Incoming.t -> t Deferred.Or_error.t
+end
+
+module type Resource_pool_intf = sig
+  include Resource_pool_base_intf
+
+  module Diff : Resource_pool_diff_intf with type pool := t
```

Comment:
Nice, I like how you refactored the code by putting `Diff` inside the actual pool.